File: pippip-native-ios/More/PublicIdCell.m
//
//  PublicIdCell.m
//  pippip-native-ios
//
//  Created by Steve Brenneis on 3/1/18.
//  Copyright © 2018 seComm. All rights reserved.
//

#import "PublicIdCell.h"
#import "ApplicationSingleton.h"
#import "CopyableLabel.h"

@interface PublicIdCell ()

@property (weak, nonatomic) IBOutlet UILabel *publicIdLabel;

@end;

@implementation PublicIdCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code

    [self attachTapHandler];
    _publicIdLabel.text = [ApplicationSingleton instance].accountSession.sessionState.publicId;

}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

- (void)attachTapHandler {
    
    [self setUserInteractionEnabled:YES];
    UIGestureRecognizer *tapped =
        [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleTap:)];
    [self addGestureRecognizer:tapped];
    
}

- (void)handleTap:(UIGestureRecognizer*)gesture {
    
    [self becomeFirstResponder];
    UIMenuController *menu = [UIMenuController sharedMenuController];
    [menu setTargetRect:self.frame inView:self.superview];
    [menu setMenuVisible:YES animated:YES];
    
}

- (void)copy:(id)sender {
    
    UIPasteboard *paste = [UIPasteboard generalPasteboard];
    paste.string = _publicIdLabel.text;
    
}

- (BOOL)canPerformAction:(SEL)action withSender:(id)sender {
    
    return action == @selector(copy:);
    
}

- (BOOL)canBecomeFirstResponder {
    return YES;
}

@end

File: pippip-native-ios/More/PublicIdCell.swift
//
//  PublicIdCell.swift
//  pippip-native-ios
//
//  Created by Steve Brenneis on 3/22/18.
//  Copyright © 2018 seComm. All rights reserved.
//

import UIKit

@objc class PublicIdCell: UITableViewCell {

    @IBOutlet weak var publicIdLabel: UILabel!

    override public var canBecomeFirstResponder: Bool {
        get {
            return true
        }
    }

    @objc class func cellItem() -> MoreCellItem {

        let item: MoreCellItem = MoreCellItem()
        item.cellHeight = 65.0
        item.cellReuseId = "PublicIdCell"
        return item

    }

    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
        publicIdLabel.text = ApplicationSingleton.instance().accountSession.sessionState.publicId
        attachTapHandler()

    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

     func attachTapHandler() {

        let tapped: UIGestureRecognizer = UILongPressGestureRecognizer(target: self,
                                                                       action: #selector(handleTap))
        addGestureRecognizer(tapped)
        
    }

    @objc func handleTap(gesture:UIGestureRecognizer) {

        becomeFirstResponder()
        let menu = UIMenuController.shared
        menu.setTargetRect(self.frame, in:self.superview!)
        menu.setMenuVisible(true, animated:true)

    }

    override func canPerformAction(_ action: Selector, withSender sender: Any?) -> Bool {

        return action == #selector(copy(_:))

    }

    override func copy(_ sender: Any?) {

        let paste = UIPasteboard.general
        paste.string = publicIdLabel.text

    }

}
